src/plugins/fts-squat/fts-backend-squat.c

```diff
@@ -137,6 +147,9 @@ fts_backend_squat_set_box(struct squat_fts_backend *backend,
 	if (backend->full_len != 0)
 		squat_trie_set_full_len(backend->trie, backend->full_len);
 	backend->box = box;
+	ret = squat_trie_open(backend->trie);
+	if (ret < 0)
+		return -1;
```

Comment:
This is intended. If ret >= 0 the next line will return what the original code did:

return squat_trie_get_last_uid(backend->trie, last_uid_r);

Previously, squat_trie_open could fail and squat_trie_get_last_uid would still be called, originating the error.

Hope this helps...